code/go/0chain.net/smartcontract/dbs/event/process.go

```diff
@@ -636,6 +636,9 @@ func (edb *EventDb) addStat(event Event) (err error) {
 		if !ok {
 			return ErrInvalidEventData
 		}
+		for _, j := range *dps {
+			j.StakedAt = time.Now()
```

Comment:
We need to use the txn.CreationDate instead of the server's time. Please update the `j.StakedAt` along with the code where other fields were set, where you will also find the `txn.CreationDate`.

Reply:
Okay.
I made the changes. Plz review